Corrected floating button in VBA flow. Now button shows at bottom of each step.

PrashantMangukiya opened the pull request:
@roryabraham pull request ready for review, if anything missing then kindly inform me, I will update it.

### Details
Corrected floating button in VBA flow, so now button shows at bottom of each page content end. So for long form user can focus on fill up information and see  button when reach at the bottom.  Only Ui related changes done without affecting logical part. i.e. moved floating button to bottom of each page within a VBA flow. 

### Fixed Issues
$ https://github.com/Expensify/App/issues/5183

### Tests
1. Create Workspace
2. Select Get Started
3. Choose Connect Manually option to add bank account via routing/account number, continue
4. See the Save & Continue option regardless of scroll position on the Company Information step
To check ui  for individual page within flow, we can open related route from localhost. 

### QA Steps
Each page within VBA flow have floating button moved to end of form/content so each page look consistent. At present some issue to fillup the form and test vba steps, so each page browsed from localhost via route and checked that  button correctly at bottom of page content.


### Tested On

- [X] Web
- [X] Mobile Web
- [X] Desktop
- [X] iOS
- [X] Android


### Screenshots
<!-- Add screenshots for all platforms tested. Pull requests won't be merged unless the screenshots show the app was tested on all platforms.-->

#### Web
<img width="1174" alt="Web1" src="https://user-images.githubusercontent.com/7823358/133322648-07db4901-d51a-4b09-8c10-25f41b1526fc.png">

<img width="1174" alt="Web2A" src="https://user-images.githubusercontent.com/7823358/133322667-da3fd084-6b79-4a73-af0b-1f40ceaec470.png">

<img width="1174" alt="Web2B" src="https://user-images.githubusercontent.com/7823358/133322709-90568a86-9f9d-4de2-84d9-47714e1cf92e.png">

<img width="1174" alt="Web3" src="https://user-images.githubusercontent.com/7823358/133322797-4029cb16-9789-43f9-acee-82c6e685cbe0.png">

<img width="1174" alt="Web4" src="https://user-images.githubusercontent.com/7823358/133322830-c1f536d6-553c-444e-96b0-9d9e634be0a0.png">


#### Mobile Web
<img width="600" alt="MobileWeb1" src="https://user-images.githubusercontent.com/7823358/133322920-3b9507c5-0813-46cb-a72e-290e8e94146d.png">

<br /><br />

<img width="600" alt="MobileWeb2A" src="https://user-images.githubusercontent.com/7823358/133323010-67ff279e-6883-48c7-b2f0-20013afdd972.png">

<br /><br />
<img width="600" alt="MobileWeb2B" src="https://user-images.githubusercontent.com/7823358/133323146-8883acd7-72ef-4147-bdfe-b0b188b1fa7f.png">

<br /><br />
<img width="600" alt="MobileWeb4" src="https://user-images.githubusercontent.com/7823358/133323316-b35e176d-91f9-46cb-b43a-85113ff90f9f.png">

#### Desktop
<img width="1312" alt="Desktop1" src="https://user-images.githubusercontent.com/7823358/133347217-6feff1fe-f79a-441a-8624-e1f3f24090db.png">

<img width="1312" alt="Desktop2A" src="https://user-images.githubusercontent.com/7823358/133347257-e951e178-b6a5-4181-bdd5-acb950852b62.png">

<img width="1312" alt="Desktop2B" src="https://user-images.githubusercontent.com/7823358/133347289-0b971996-7227-4d10-ab69-33e92ae50337.png">

<img width="1312" alt="Desktop3" src="https://user-images.githubusercontent.com/7823358/133347310-07fa436f-81ca-4076-879a-a58f9dbf5c5c.png">

<img width="1312" alt="Desktop4" src="https://user-images.githubusercontent.com/7823358/133347338-52856d88-e2ce-4f8d-9a6d-8a5e3e62d0ba.png">

#### iOS
<img width="478" alt="iOS1" src="https://user-images.githubusercontent.com/7823358/133347569-07d52e4c-ee40-401f-9553-57f953faa5ba.png">

<img width="478" alt="iOS2A" src="https://user-images.githubusercontent.com/7823358/133347611-da81ea9f-3dc2-405e-8a85-704d0b8404dc.png">

<img width="478" alt="iOS2B" src="https://user-images.githubusercontent.com/7823358/133347633-434e9397-ccb0-4153-ab77-41a0fafd4989.png">

<img width="478" alt="iOS3" src="https://user-images.githubusercontent.com/7823358/133347653-b40a8620-a9bb-4dea-b4a7-06f11ef7c61f.png">

<img width="478" alt="iOS4" src="https://user-images.githubusercontent.com/7823358/133347921-3b330220-2af5-47ae-8796-67e8fac36b44.png">

#### Android

<img width="600" alt="Android1" src="https://user-images.githubusercontent.com/7823358/133347960-084b0eb6-b449-4f45-8227-5657bd97db8a.png">

<img width="600" alt="Android2A" src="https://user-images.githubusercontent.com/7823358/133348043-488b22be-e5e8-4f9b-97e3-b3662fea6699.png">

<img width="600" alt="Android2B" src="https://user-images.githubusercontent.com/7823358/133348078-bb1c6354-49ec-4baf-b405-268447cd8372.png">

<img width="600" alt="Android3" src="https://user-images.githubusercontent.com/7823358/133348128-19e936ef-8a01-4e71-8466-c8f5c8379e51.png">

<img width="600" alt="Android4" src="https://user-images.githubusercontent.com/7823358/133348163-684458c1-b72e-46f8-9e31-f576198190a5.png">

PrashantMangukiya committed:
Corrected floating button in VBA flow. Now button shows at bottom of each step.

roryabraham commented:
@Expensify/design Should the button be on the bottom of the screen, even if you have to scroll down to the bottom to see it?

Also @PrashantMangukiya I don't think I'll be able to review this until Sunday or Monday.

michelle-thompson commented:
Yes, the button should be bottom-docked

shawnborton commented:
I agree with Michelle. We can probably accomplish this with some flex grow magic is my guess.

roryabraham commented:
Thanks @michelle-thompson and @shawnborton, that was my thought too!

@PrashantMangukiya Let us know if you have any questions.

marcaaron commented:
Just to confirm we are asking that if the entire form only takes up the partial height of the screen push the button to the bottom? Otherwise if the form takes up more than the full height of the screen then the button should already be at the bottom when the form is scrolled?

marcaaron commented on src/pages/ReimbursementAccount/BankAccountStep.js:
```diff
@@ -298,14 +298,14 @@ class BankAccountStep extends React.Component {
+                                isDisabled={!this.canSubmitManually()}
+                                onPress={this.addManualAccount}
+                            />
                         </View>
```
I know there's not a lot going on in this step - but I think this should be a `ScrollView`?

PrashantMangukiya replied:
Replaced View with ScrollView as per request, also adjusted style accordingly.

shawnborton commented:
Yup, that sounds right @marcaaron

marcaaron commented on src/pages/ReimbursementAccount/ValidationStep.js:
```diff
@@ -141,7 +141,7 @@ class ValidationStep extends React.Component {
                             </Text>
                         </View>
-                        <View style={[styles.m5, styles.flex1]}>
+                        <View style={[styles.m5]}>
```
Also feels like this should be a `ScrollView` for consistency with the other forms.

PrashantMangukiya replied:
Thanks you for request. I set ScrollView at line 135 instead of 144. Because above the line 144, some information is displayed in a sibling View at line 136-143 (validationStep.description and validationStep.descriptionCTA) It is also part of this page/form, So view at line 135 is parent of both ( i.e. view at 136, and 144) so I put ScrollView at parent level view i.e. at line 135 So form instruction and input field both comes within ScrollView and scroll both together and page behave normally.

PrashantMangukiya committed:
Replaced View with ScrollView within BankAccountStep.js and ValidationStep.js

PrashantMangukiya commented:
Hi @marcaaron  I pushed new commit that includes changes you have suggested. Thank you for your time.

roryabraham commented:
@PrashantMangukiya It seems like you haven't yet made this change:

> if the entire form only takes up the partial height of the screen, push the button to the bottom. Otherwise, if the form takes up more than the full height of the screen, then the button should already be at the bottom when the form is scrolled.

Please update the PR screenshots and let us know when that change has been made.

PrashantMangukiya commented:
@roryabraham, Sorry for confusion to me, I though we are discussing and not finalised yet, so I did not implemented. I am making changes accordingly and submitting new commit and update pr screenshot soon. Thanks.

marcaaron commented:
@PrashantMangukiya I think this solution might work:

- Make the `ScrollView` have styles like this:

```jsx
            <ScrollView
                style={[styles.w100, styles.flex1]}
                contentContainerStyle={styles.flexGrow1}
            >
```

- The button can then be wrapped in a `View` with a style of `flex1`.

FWIW I solved it this way in another PR [here](https://github.com/Expensify/App/pull/5213/files) that we will merge after this one (but haven't yet tested the solution on native platforms)

PrashantMangukiya committed:
Pushed button at bottom if partial content height, otherwise button shows at bottom of content

If the entire form only takes up the partial height of the screen, push the button to the bottom. Otherwise, if the form takes up more than the full height of the screen, then the button should already be at the bottom when the form is scrolled.

PrashantMangukiya commented:
@roryabraham Changes made as requested i.e. New commit pushed, and pr screenshot also updated.  @marcaaron thank you for suggestion. You can proceed for pr review.

marcaaron reviewed (changes requested):
Please test on Android & iOS and provide screenshots. I don't see any reason to mark these as `N/A`

PrashantMangukiya commented:
@marcaaron as mentioned in the https://github.com/Expensify/App/issues/5183#issuecomment-917141300  vba flow cant run via form input at present and it has some open issue etc. So not able  to fill up form and  run complete flow in iOS and android, so screenshot not produced. So due to that limitation we used direct link as shown below to test updated ui at locally. Because in this issue only button related ui changes was requested.

- http://localhost:8080/bank-account/new
- http://localhost:8080/bank-account/company
- http://localhost:8080/bank-account/personal-information
- http://localhost:8080/bank-account/contract
- http://localhost:8080/bank-account/validate

marcaaron commented:
Try using deep links?

marcaaron commented:
I think you can just send yourself a message like this then click on it and it should open

<img width="464" alt="2021-09-14_12-24-45" src="https://user-images.githubusercontent.com/32969087/133341858-53712ab6-d3f9-445a-b5f7-672670323a59.png">

PrashantMangukiya commented:
@marcaaron , thanks for suggesting deep link, let me try that way to create screenshot for iOS and Android if I can do. Thanks.

PrashantMangukiya commented:
@marcaaron, Attached is the screen shot for iOS, Android and Desktop. Thank you again for deep link suggestion.

marcaaron reviewed (approved):
This LGTM. Normally I might ask to see if we can reduce some duplication around the styles in the `ScrollView`, but I'm already doing this work in another issue so I think what we have here is 👌  for now.

roryabraham reviewed (approved):
LGTM 👍

OSBotify commented:
:hand: This PR was not deployed to staging yet because QA is ongoing. It will be automatically deployed to staging after the next production release.

OSBotify commented:
🚀 [Deployed](https://github.com/Expensify/App/actions/runs/1239082061) to staging by @roryabraham in version: 1.0.98-2 🚀

 platform | result 
 ---|--- 
🤖 android 🤖|success ✅ 
🖥 desktop 🖥|success ✅
🍎 iOS 🍎|success ✅ 
🕸 web 🕸|success ✅

OSBotify commented:
🚀 [Deployed](https://github.com/Expensify/App/actions/runs/1255109644) to production by @Jag96 in version: 1.0.99-4 🚀

 platform | result 
 ---|--- 
🤖 android 🤖|success ✅ 
🖥 desktop 🖥|success ✅
🍎 iOS 🍎|success ✅ 
🕸 web 🕸|success ✅
